lib/ProviderAws.js

```diff
@@ -242,32 +250,132 @@ module.exports = function(S) {
+      return path.join(configDir, 'config');
+    }
+
+    /**
      * Get All Profiles
-     * - Gets all profiles from ~/.aws/credentials
+     * - Gets all profiles from AWS credentials, config file
      */
 
     getAllProfiles() {
-      let credsPath = path.join(this.getConfigDir(), 'credentials');
+      let credsPath = this.getAwsCredentialsFile();
+      let configPath = this.getAwsConfigFile();
+
+      let creds;
       try {
-        return AWS.util.ini.parse(AWS.util.readFileSync(credsPath));
+        creds = AWS.util.ini.parse(AWS.util.readFileSync(credsPath));
       }
       catch (e) {
-        return null;
+        creds = {};
       }
+
+      let configs;
+      try {
+        configs = AWS.util.ini.parse(AWS.util.readFileSync(configPath));
+      }
+      catch (e) {
+        configs = {};
+      }
+
+      // First, load up all profile from config file.
+      var profiles = Object.keys(configs).reduce((obj, key) => {
+        const match = key.match(/^profile (.+)/);
+        if (match) {
+          obj[match[1]] = configs[key];
+        }
+        return obj;
+      }, {});
+
+      // Now, load profiles from credentials file (overriding any values found in config file)
+      Object.keys(creds).forEach((name) => {
+        profiles[name] = Object.assign(profiles[name] || {}, creds[name]);
+      });
+
+      return profiles;
     }
 
     /**
      * Get Profile
-     * - Gets a single profile from ~/.aws/credentials
+     * - Gets a single profile from AWS credentials, config files.
      */
 
     getProfile(awsProfile, optional) {
       let profile = awsProfile.toLowerCase(),
         profiles = this.getAllProfiles();
-      if (!optional && !profiles[profile]) {
-        throw new SError(`Cant find profile ${profile} in ~/.aws/credentials`, profile);
+      let profileConfig = profiles[profile];
+      if (!optional && !profileConfig) {
+        throw new SError(`Cant find profile ${profile} in AWS credential file and/or AWS config file`, profile);
       }
-      return profiles[profile];
+
+      if (profileConfig && profileConfig.source_profile && profileConfig.role_arn) {
+        return this.getRoleCredentials(profile, profiles);
+      } else {
+        return BbPromise.resolve(profileConfig);
+      }
+    }
+
+    /**
+     * Get Role Credentials
+     * - Gets temporary credentials via assuming a role
+     */
+    getRoleCredentials(profile, profiles) {
+      const sourceProfile = profiles[profile].source_profile;
+      const roleArn = profiles[profile].role_arn;
+      const sourceProfileCredentials = profiles[sourceProfile];
+      if (!sourceProfileCredentials) {
+        throw new SError(`Cant find source profile ${sourceProfile} in AWS credential file and/or AWS config file`, sourceProfile);
+      }
+
+      const sourceAccessKeyId = sourceProfileCredentials.aws_access_key_id;
+      const sourceSecretAccessKey = sourceProfileCredentials.aws_secret_access_key;
```

Comment:
This doesn't allow for tokens, see lines 146 and 149.  Probably worth consolidating that logic of extracting credentials from a profile so that we only define the attributes to look for once.